smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java

```diff
@@ -74,12 +74,26 @@ public static SMSHandler getInstance() {
     /**
      * Setup for the handler.
      *
-     * @param context current context.
-     */
-    public void setup(Context context) {
+     * @param context current context. Must not be null
+     * @throws NullPointerException If the given context is null
+     */
+    public void setup(@NonNull Context context) {
+        /*
+        null pointer exception is thrown because if the
+        context is null, this.context isn't, it's a WeakReference to a null object,
+        which later on causes problems
+         */
+        if(context == null) throw new NullPointerException("The given context must not be null!");
         this.context = new WeakReference<>(context);
     }
```

Comment:
Lo commento per ricordarmi di fare ricerche a riguardo

Reply:
@EnricoCestaro did you check this out yet?